channeldb/invoices.go

```diff
@@ -1241,79 +1260,79 @@ func (d *DB) updateInvoice(hash lntypes.Hash, invoices, settleIndex *bbolt.Bucke
 		return &invoice, err
 	}
 
-	// Update invoice state.
-	invoice.State = update.State
+	// If there is nothing to update, return early.
+	if update == nil {
```

Comment:
Everywhere we return `nil` in `InvoiceRegistry.updateInvoice`. For example an htlc that has the wrong amount. We execute the update logic and conclude we don't need to update the invoice.

Reply:
It is documented on the field description of `InvoiceUpdateDesc`

Reply:
That is for the `State` field, not the `InvoiceStateUpdateDesc` itself, which is what we check here. I think it would make sense for the callback function documentation to note this.